common/Geometry2d/Segment.cpp

```diff
@@ -18,6 +18,13 @@ Rect Segment::bbox() const {
 }
 
 float Segment::distTo(const Point& other) const {
+    float dist = nearestPoint(other).distTo(other);
+    if (nearlyEqual(dist, 0)) {
```

Comment:
Any reason for the special case for zero here?  We might as well just return `dist` always.